Clicker/Models/Socket.swift

```diff
@@ -7,88 +7,82 @@
 //
 
 import SocketIO
+import SwiftyJSON
 
 class Socket {
     let id: String
-    var delegates: [SocketDelegate] = [SocketDelegate]()
+    var delegate: SocketDelegate
     var socket: SocketIOClient
     var manager: SocketManager
     
-    init(id: String, userType: String) {
+    init(id: String, userType: String, delegate: SocketDelegate) {
```

Comment:
I think it would be more readable to have `userType: UserType` and convert `userType` to a string in `init`, rather than having `userType: string` and converting to a string elsewhere before initializing a `Socket`